src/entities/Proposal/routes.ts

```diff
@@ -520,6 +574,37 @@ export async function proposalComments(req: Request<{ proposal: string }>){
   }
 }
 
+async function getPassedProposals(req:Request<{proposal_type: ProposalType}>){
+  const proposalType = req.params.proposal_type
+  let passedProposals = await ProposalModel.getPassedProposals(proposalType)
+  return passedProposals.map(proposal => {
+    return {
+      key: proposal.id,
+      text: proposal.title,
+      value: proposal.id
+    }
+  })
+}
+
+async function validateLinkedProposal(linkedProposalId: string, expectedProposalType: ProposalType) {
+  const linkedProposal = await ProposalModel.findOne<ProposalAttributes>({
+    id: linkedProposalId,
+    type: expectedProposalType,
+    deleted: false
+  })
+  if (!linkedProposal) {
+    throw new RequestError(`Could not find a matching ${expectedProposalType} proposal for "${linkedProposalId}"`, RequestError.NotFound)
+  }
+}
+
+async function validateSubmissionThreshold(user: string, submissionThreshold?: string) {
+  const requiredVp = Number(submissionThreshold || requiredEnv('SUBMISSION_THRESHOLD_POLL'))
```

Comment:
the function `requiredEnv` should be use on top level, its goal is to know at bootstrap time if there is a missing ENV